### PR TITLE
source-float contribution from bala-ceg

### LINKED ISSUE #45988: [connector-request] Float
This is a **suggestion** to create a connector for **Float** using the UI Connector Builder.
The website for the connector is https://www.floatapp.com.

If you want to try, claim this issue and start working on it following the steps below.

### Steps:
1. Comment in the issue and wait to be assigned to start working on it.
2. Map the API endpoints (get approval to move to 3 step)
3. Describe steps to get credentials.
4. Create the connector using the UI Builder

> [!IMPORTANT]
> Airbyte doesn't have sandbox credentials for this connector. You must have access/credentials to the service provider to create the connector.

> [!CAUTION]
> This is a **suggestion** and there may be cases where creating the connector using the Builder won't be possible. To minimize the risk of investing a lot of time directly in creating the connector, we strongly recommend to follow steps above.

### PR DESCRIPTION
**Pull request by bala-ceg**
## What

This PR adds a new connector for source float (source-float).

Contributor-provided connector description:

source-float airbyte connector is designed to sync scheduling and resource management data between Float, a leading resource management platform, and external systems. 

Float.com enables teams to plan and allocate resources effectively, manage team availability, and track project timelines. This connector automates the data flow between Float and other platforms, ensuring that resource schedules and project plans are up-to-date across all tools you use.
## Details

<details><summary>Expand for list of spec properties and streams</summary>

### Configuration

| Input | Type | Description | Default Value |
|-------|------|-------------|---------------|
| `access_token` | `string` | Float Access Token. API token obtained from your Float Account Settings page |  |

### Streams
| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
|-------------|-------------|------------|---------------------|----------------------|
| accounts | account_id | DefaultPaginator | ✅ |  ❌  |
| departments | department_id | DefaultPaginator | ✅ |  ❌  |
| people | people_id | DefaultPaginator | ✅ |  ❌  |
| roles | id | DefaultPaginator | ✅ |  ❌  |
| public_holidays | id | DefaultPaginator | ✅ |  ❌  |
| holidays | holiday_id | DefaultPaginator | ✅ |  ❌  |
| projects | project_id | DefaultPaginator | ✅ |  ❌  |
| status | status_id | DefaultPaginator | ✅ |  ❌  |
| time_off | timeoff_id | DefaultPaginator | ✅ |  ❌  |
| timeoff-types | timeoff_type_id | DefaultPaginator | ✅ |  ❌  |
| clients | client_id | DefaultPaginator | ✅ |  ❌  |
| phases | phase_id | DefaultPaginator | ✅ |  ❌  |
| project-tasks | task_meta_id | DefaultPaginator | ✅ |  ❌  |
| milestones | milestone_id | DefaultPaginator | ✅ |  ❌  |
| tasks | task_id | DefaultPaginator | ✅ |  ❌  |
| logged-time | logged_time_id | DefaultPaginator | ✅ |  ❌  |

</details>

## Reviewer checklist
- [ ] Run `/format-fix` to resolve any formatting errors
- [ ] Help the contributor upload an icon that follows our icon design guidelines
- [ ] Click into the CI workflows that wait for a maintainer to run them, which should trigger CI runs

### CODE REVIEW

File: airbyte-integrations/connectors/source-float/manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,1155 @@
...
+
+spec:
+  type: Spec
+  connection_specification:
+    type: object
+    $schema: http://json-schema.org/draft-07/schema#
+    required:
+      - float_access_token
+    properties:
+      float_access_token:
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-float/manifest.yaml:614]: Rename to `access_token`

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Small changes connector looks good, can you upload the icon?

**Comment by marcosmarxm** [comment]
Hello @bala-ceg please check my review :) contribution is almost finish.

File: airbyte-integrations/connectors/source-float/manifest.yaml (hunk 1 above)
Comment (bala-ceg) [reply airbyte-integrations/connectors/source-float/manifest.yaml]: Changed the name to access_token

**Comment by bala-ceg** [comment]
@marcosmarxm  uploaded the icon.svg and the connector is ready for your review

File: airbyte-integrations/connectors/source-float/manifest.yaml [hunk 2]
```diff
@@ -0,0 +1,1155 @@
...
+  allocations:
+    type: object
+    $schema: http://json-schema.org/draft-07/schema#
+    additionalProperties: true
+    properties: {}
+  logged-time:
+    type: object
+    $schema: http://json-schema.org/draft-07/schema#
+    additionalProperties: true
+    properties: {}
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-float/manifest.yaml:1155]: @bala-ceg can you check why these streams aren't with the correct schema?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
`allocations` let's rename to `tasks` makes more sense. The `task_id` is the unique identifier of this stream.

Incremental: some endpoints support incremental, do you want to implement them?

Icon: it must be a square svg icon.

**Comment by marcosmarxm** [comment]
@bala-ceg did you check my last review? 

I created this [view](https://github.com/orgs/airbytehq/projects/104/views/3) to help contributors know when it will happen the new review. I'm moving Float next review to Friday (2024-10-25)

**Comment by marcosmarxm** [comment]
Hello 👋 Due to the numerous contributions happening, I decided to create this [view](https://github.com/orgs/airbytehq/projects/104/views/3) to help you know when I plan to review your contribution. If your connector has some limited-time credential feel free to reach me out.

File: airbyte-integrations/connectors/source-float/manifest.yaml (hunk 2 above)
Comment (bala-ceg) [reply airbyte-integrations/connectors/source-float/manifest.yaml]: I checked and fixed the streams with incorrect schema

**Review by marcosmarxm — APPROVED** [review]
Great contribution @bala-ceg

**Pull request merged**